From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sun, 15 Jul 2018 22:31:37 +0900
Subject: sysusers,tmpfiles: re-create systemd-network,
 systemd-resolve and systemd-timesync

This partially reverts d4e9e574ea0b5d23598a317e68399584d229568b,
0187368cadea183e18c6d575a9d6b7f491a402af, and
4240cb02fda90ba11dfc0114201e42691132c6a9.
The services systemd-networkd, systemd-resolved, and systemd-timesyncd
enable DynamicUsers= and have bus interfaces. Unfortunately, these
has many problems now. Let us create the relevant users, at least,
tentatively.

Fixes #9503.

(cherry picked from commit 5b5d82615011b9827466b7cd5756da35627a1608)
---
 sysusers.d/systemd.conf.m4 | 9 +++++++++
 tmpfiles.d/systemd.conf.m4 | 6 +++---
 2 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/sysusers.d/systemd.conf.m4 b/sysusers.d/systemd.conf.m4
index 2315b56..ef5a3cb 100644
--- a/sysusers.d/systemd.conf.m4
+++ b/sysusers.d/systemd.conf.m4
@@ -6,6 +6,15 @@
 #  (at your option) any later version.
 
 g systemd-journal   - -
+m4_ifdef(`ENABLE_NETWORKD',
+u systemd-network   - "systemd Network Management"
+)m4_dnl
+m4_ifdef(`ENABLE_RESOLVE',
+u systemd-resolve   - "systemd Resolver"
+)m4_dnl
+m4_ifdef(`ENABLE_TIMESYNCD',
+u systemd-timesync  - "systemd Time Synchronization"
+)m4_dnl
 m4_ifdef(`ENABLE_COREDUMP',
 u systemd-coredump  - "systemd Core Dumper"
 )m4_dnl
diff --git a/tmpfiles.d/systemd.conf.m4 b/tmpfiles.d/systemd.conf.m4
index 120e42a..d39c9cb 100644
--- a/tmpfiles.d/systemd.conf.m4
+++ b/tmpfiles.d/systemd.conf.m4
@@ -17,9 +17,9 @@ d /run/systemd/users 0755 root root -
 d /run/systemd/machines 0755 root root -
 d /run/systemd/shutdown 0755 root root -
 m4_ifdef(`ENABLE_NETWORKD',
-d /run/systemd/netif 0755 - - -
-d /run/systemd/netif/links 0755 - - -
-d /run/systemd/netif/leases 0755 - - -
+d /run/systemd/netif 0755 systemd-network systemd-network -
+d /run/systemd/netif/links 0755 systemd-network systemd-network -
+d /run/systemd/netif/leases 0755 systemd-network systemd-network -
 )m4_dnl
 
 d /run/log 0755 root root -
